lsp-mode.el

```diff
@@ -634,6 +634,15 @@ diagnostics have changed."
   :type 'hook
   :group 'lsp-mode)
 
+(defcustom lsp-diagnostics-default-severity 'error
+  "Error level to use when the server does not report one."
+  :type '(choice (const :tag "Error"       error)
+                 (const :tag "Warning"     warning)
+                 (const :tag "Information" info)
+                 (const :tag "Hint"        hint))
+  :group 'lsp-mode
+  :package-version '(lsp-mode . "9.0.1"))
```

Comment:
Should we keep this in `lsp-diagnostics.el`? 🤔

Reply:
Hi @jcs090218, the reason I put it in lsp-mode.el is because there are multiple users, not just lsp-diagnostics.el.  lsp-modeline, lsp-headerline, lsp-mode and lsp-diagnostics all need this information, so it seemed more general and likely belongs in lsp-mode.el.  It answers the general question of what the user wants to do in the absence of LSP severity information.  lsp-diagnostics.el is just one such user of this information.

I placed it in lsp-mode.el near a couple other diagnostic-related user options (i.e., lsp-after-diagnostics-hook and lsp-diagnostics-updated-hook) to be consistent there.  It appears that lsp-diagnostics.el is strictly the diagnostics provider (i.e., Flymake/Flycheck) interfacing, but other LSP diagnostics related information is kept in lsp-mode.el.